tests/phpunit/test-class-wc-stripe-upe-payment-gateway.php

```diff
@@ -156,6 +156,83 @@ private function get_order_details( $order ) {
 		return [ $amount, $description, $metadata ];
 	}
 
+	/**
+	 * @dataProvider get_upe_available_payment_methods_provider
+	 */
+	public function test_get_upe_available_payment_methods( $country, $available_payment_methods ) {
+		$this->set_stripe_account_data( [ 'country' => $country ] );
+		$this->assertSame( $available_payment_methods, $this->mock_gateway->get_upe_available_payment_methods() );
+	}
+
+	/**
+	 * @dataProvider get_upe_enabled_at_checkout_payment_method_ids_provider
+	 */
+	public function test_get_upe_enabled_at_checkout_payment_method_ids( $country, $available_payment_methods ) {
+		$this->set_stripe_account_data( [ 'country' => $country ] );
+		$this->mock_gateway->update_option(
+			'upe_checkout_experience_accepted_payments',
+			[
+				'card',
+				'link',
+			]
+		);
+		$this->assertSame( $available_payment_methods, $this->mock_gateway->get_upe_enabled_at_checkout_payment_method_ids() );
+	}
+
+	public function get_upe_enabled_at_checkout_payment_method_ids_provider() {
+		return [
+			[
+				'US',
+				[
+					WC_Stripe_UPE_Payment_Method_CC::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Link::STRIPE_ID,
+				],
+			],
+			[
+				'NON_US',
+				[
+					WC_Stripe_UPE_Payment_Method_CC::STRIPE_ID,
+				],
+			],
+		];
+	}
+
+	public function get_upe_available_payment_methods_provider() {
+		return [
+			[
+				'US',
+				[
+					WC_Stripe_UPE_Payment_Method_CC::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Giropay::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Eps::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Bancontact::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Boleto::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Ideal::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Oxxo::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Sepa::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_P24::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Sofort::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Link::STRIPE_ID,
+				],
+			],
+			[
+				'NON_US',
+				[
+					WC_Stripe_UPE_Payment_Method_CC::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Giropay::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Eps::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Bancontact::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Boleto::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Ideal::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Oxxo::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Sepa::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_P24::STRIPE_ID,
+					WC_Stripe_UPE_Payment_Method_Sofort::STRIPE_ID,
+				],
+			],
+		];
```

Comment:
Perhaps here we could use `WC_Stripe_UPE_Payment_Gateway::UPE_AVAILABLE_METHODS` for the US and `array_filter` out `'link' === STRIPE_ID` for non-US. Would be slightly lower maintenance when adding new payment methods.

Very minor suggestion, so feel free to ignore 🙂.

Reply:
`WC_Stripe_UPE_Payment_Gateway::UPE_AVAILABLE_METHODS` contains the class names, and we need the Stripe id as `get_upe_available_payment_methods` returns the payment method id.

Reply:
True.